Trying to add disposable to a disposable store that has already been disposed of

Saw while playing around in code OSS

<img width="1251" alt="image" src="https://user-images.githubusercontent.com/4544166/216435879-aa961ad9-944a-49c2-a426-64fd3eb1117c.png">

I also just ran into this. Here is the complete stack trace:
```
lifecycle.js:207 Error: Trying to add a disposable to a DisposableStore that has already been disposed of. The added object will be leaked!
    at DisposableStore.add (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/base/common/lifecycle.js:207:34)
    at CustomizedMenuWorkbenchToolBar.setActions (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/platform/actions/browser/toolbar.js:108:42)
    at CustomizedMenuWorkbenchToolBar.updateToolbar (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/editor/contrib/inlineCompletions/browser/inlineSuggestionHintsWidget.js:212:18)
    at CustomizedMenuWorkbenchToolBar.setAdditionalSecondaryActions (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/editor/contrib/inlineCompletions/browser/inlineSuggestionHintsWidget.js:216:18)
    at InlineSuggestionHintsContentWidget.update (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/editor/contrib/inlineCompletions/browser/inlineSuggestionHintsWidget.js:155:26)
    at InlineSuggestionHintsWidget.update (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/editor/contrib/inlineCompletions/browser/inlineSuggestionHintsWidget.js:35:29)
    at vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/editor/contrib/inlineCompletions/browser/inlineSuggestionHintsWidget.js:29:71
    at Listener.invoke (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/base/common/event.js:530:27)
    at EventDeliveryQueue.deliver (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/base/common/event.js:705:38)
    at Emitter.fire (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/base/common/event.js:670:37)
    at vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/editor/browser/widget/codeEditorWidget.js:135:48
    at Listener.invoke (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/base/common/event.js:530:27)
    at PrivateEventDeliveryQueue.deliver (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/base/common/event.js:705:38)
    at Emitter.fire (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/base/common/event.js:670:37)
    at EditorConfiguration._recomputeOptions (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/editor/browser/config/editorConfiguration.js:55:31)
    at EditorConfiguration.updateOptions (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/editor/browser/config/editorConfiguration.js:102:18)
    at CodeEditorWidget.updateOptions (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/editor/browser/widget/codeEditorWidget.js:241:33)
    at TextResourceEditor.updateEditorControlOptions (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/workbench/browser/parts/editor/textCodeEditor.js:33:33)
    at TextResourceEditor.updateEditorConfiguration (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/workbench/browser/parts/editor/textEditor.js:177:22)
    at vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/workbench/browser/parts/editor/textEditor.js:111:72
    at Listener.invoke (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/base/common/event.js:530:27)
    at EventDeliveryQueue.deliver (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/base/common/event.js:705:38)
    at Emitter.fire (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/base/common/event.js:670:37)
    at CodeEditorWidget.setModel (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/editor/browser/widget/codeEditorWidget.js:308:36)
    at vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/editor/browser/widget/codeEditorWidget.js:1145:67
    at Listener.invoke (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/base/common/event.js:530:27)
    at PrivateEventDeliveryQueue.deliver (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/base/common/event.js:705:38)
    at Emitter.fire (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/base/common/event.js:670:37)
    at TextModel.dispose (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/editor/common/model/textModel.js:218:33)
    at WorkbenchModelService.destroyModel (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/editor/common/services/modelService.js:388:29)
    at UntitledTextEditorModel.dispose (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/workbench/common/editor/textEditorModel.js:179:35)
    at vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/workbench/services/textfile/common/textEditorService.js:90:81
    at vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/base/common/event.js:76:37
    at Listener.invoke (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/base/common/event.js:530:27)
    at PrivateEventDeliveryQueue.deliver (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/base/common/event.js:705:38)
    at Emitter.fire (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/base/common/event.js:670:37)
    at UntitledTextEditorInput.dispose (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/workbench/common/editor/editorInput.js:225:37)
    at UntitledTextEditorInput.dispose (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/workbench/services/untitled/common/untitledTextEditorInput.js:135:19)
    at vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/workbench/services/untitled/common/untitledTextEditorInput.js:41:57
    at Listener.invoke (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/base/common/event.js:530:27)
    at PrivateEventDeliveryQueue.deliver (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/base/common/event.js:705:38)
    at Emitter.fire (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/base/common/event.js:670:37)
    at UntitledTextEditorModel.revert (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/workbench/services/untitled/common/untitledTextEditorModel.js:152:31)
    at NativeTextFileService.revert (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/workbench/services/textfile/browser/textFileService.js:501:34)
    at UntitledTextEditorInput.revert (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/workbench/common/editor/textResourceEditorInput.js:54:40)
    at EditorGroupView.doHandleCloseConfirmation (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/workbench/browser/parts/editor/editorGroupView.js:1236:38)
    at async EditorGroupView.handleCloseConfirmation (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/workbench/browser/parts/editor/editorGroupView.js:1125:24)
    at async EditorGroupView.closeEditors (vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/workbench/browser/parts/editor/editorGroupView.js:1266:26)
    at async vscode-file://vscode-app/Users/alex/src/vscode3/out/vs/workbench/browser/parts/editor/editorCommands.js:676:21
    at async Promise.all (index 0)
```

--- a/src/vs/editor/contrib/inlineCompletions/browser/inlineSuggestionHintsWidget.ts
+++ b/src/vs/editor/contrib/inlineCompletions/browser/inlineSuggestionHintsWidget.ts
@@ -36,6 +36,7 @@ export class InlineSuggestionHintsWidget extends Disposable {
 	private readonly widget = this._register(this.instantiationService.createInstance(InlineSuggestionHintsContentWidget, this.editor, true));
 
 	private sessionPosition: Position | undefined = undefined;
+	private isDisposed = false;
 
 	constructor(
 		private readonly editor: ICodeEditor,
@@ -51,7 +52,16 @@ export class InlineSuggestionHintsWidget extends Disposable {
 		this.update();
 	}
 
+	override dispose(): void {
+		this.isDisposed = true;
+		super.dispose();
+	}
+
 	private update(): void {
+		if (this.isDisposed) {
+			return;
+		}
+
 		const options = this.editor.getOption(EditorOption.inlineSuggest);
 		if (options.showToolbar !== 'always' || !this.model.ghostText) {
 			this.widget.update(null, 0, undefined, []);

